Classes/common/CDTIncrementalStore.m

```diff
@@ -0,0 +1,2398 @@
+    newRev.body[CDTISEntityNameKey] = [entity name];
+    newRev.body[CDTISIdentifierKey] = [[[mo objectID] URIRepresentation] absoluteString];
+    if ([blobStore count]) {
+        newRev.attachments = blobStore;
+    }
+
+    CDTDocumentRevision *rev = [self.datastore createDocumentFromRevision:newRev error:&err];
+    if (!rev) {
+        if (error) *error = err;
+        return NO;
+    }
+
+    if (CDTISReadItBack) {
+        /**
+         *  See CDTISReadItBack
+         */
+        rev = [self.datastore getDocumentWithId:newRev.docId error:&err];
+        if (!rev) {
+            // Always oops!
+            oops(@"ReadItBack: error: %@", err);
+        }
+        // Always oops
+        if ([rev.body count] == 0) oops(@"empty save");
+    }
+
+    return YES;
+}
+
+/**
+ *  Update existing object in the database
+ *
+ *  @param mo    Managed Object
+ *  @param error Error
+ *
+ *  @return YES/NO
+ */
+- (BOOL)updateManagedObject:(NSManagedObject *)mo error:(NSError **)error
+{
+    NSError *err = nil;
+    NSManagedObjectID *moid = [mo objectID];
+
+    if (badObjectVersion(moid, self.metadata)) oops(@"hash mismatch?: %@", moid);
+
+    NSString *docID = [self stringReferenceObjectForObjectID:moid];
+
+    NSEntityDescription *entity = [mo entity];
+    NSDictionary *propDic = [entity propertiesByName];
+
+    NSMutableDictionary *props = [NSMutableDictionary dictionary];
+    NSMutableDictionary *blobStore = [NSMutableDictionary dictionary];
+    NSDictionary *changed = [mo changedValues];
+
+    for (NSString *name in changed) {
+        NSDictionary *enc = nil;
+        id prop = propDic[name];
+        if ([prop isTransient]) {
+            continue;
+        }
+        if ([prop isKindOfClass:[NSAttributeDescription class]]) {
+            NSAttributeDescription *att = prop;
+            enc = [self encodeAttribute:att withValue:changed[name] blobStore:blobStore error:&err];
+        } else if ([prop isKindOfClass:[NSRelationshipDescription class]]) {
+            NSRelationshipDescription *rel = prop;
+            enc = [self encodeRelation:rel withValue:changed[name] error:&err];
+        } else {
+            oops(@"bad prop?");
+        }
+        [props addEntriesFromDictionary:enc];
+    }
+
+    CDTDocumentRevision *oldRev = [self.datastore getDocumentWithId:docID error:&err];
+    if (!oldRev) {
+        if (error) *error = err;
+        return NO;
+    }
+
+    // TODO: version HACK
+    NSString *oldVersion = oldRev.body[CDTISObjectVersionKey];
+    uint64_t version = [oldVersion longLongValue];
+    ++version;
+    NSNumber *v = [NSNumber numberWithUnsignedLongLong:version];
+    props[CDTISObjectVersionKey] = [v stringValue];
+
+    CDTMutableDocumentRevision *upRev = [oldRev mutableCopy];
+
+    // update attachments first
+    if ([blobStore count]) {
+        [upRev.attachments addEntriesFromDictionary:blobStore];
+    }
+
+    // delete all changed properties, in case they are being removed.
```

Comment:
***YES*** we can skip the removal of the objects.
Early in development, we sprinkled the meta-data around the document body, which could cause a stale meta-data entry to be present and stale.  This is no longer the case, and properties requiring meta always have a collected meta-data dictionary associated with them.  This results in a consistent document body that requires no clean up.
A comment will be added that explains this.